eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/AggregateCriterion.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+
+class AggregateCriterion extends CriterionHandler
```

Comment:
I don't follow naming a bit here, could you give me examples of aggregate and non-aggregate Criteria? What does it aggregate? Based on the body of `handle` method I would rather say `ComplexCriterionHandler` or `CompositeCriterionHandler`.

For `AggregateCriterion` I expected rather something like
``` php
foreach ($criterion->criteria as $nestedCriterion)  {
   // ...    
}
```
which is not the case.

Reply:
Renamed to `CompositeCriterion` in https://github.com/ezsystems/ezpublish-kernel/pull/2958/commits/f5b574533d90c7f342beefeca599f60bf1ecbdd0